velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -207,6 +229,42 @@ VeloxToSubstraitExprConvertor::toSubstraitLiteral(
   return *literalExpr;
 }
 
+void VeloxToSubstraitExprConvertor::complexVectorToLiteral(
+    google::protobuf::Arena& arena,
+    std::shared_ptr<ConstantVector<ComplexType>> constantVector,
+    ::substrait::Expression_Literal* substraitLiteral) {
+  VELOX_CHECK_EQ(
+      constantVector->size(), 1, "Only one complex vector is expected.");
+  auto arrayVector = constantVector->valueVector()->as<ArrayVector>();
+  VELOX_CHECK_NOT_NULL(
+      arrayVector,
+      "arrayVector is NULL, Maybe Substrait literal is not supported for {}.",
+      constantVector->type()->toString());
+  VELOX_CHECK_EQ(arrayVector->size(), 1, "Only one array is expected.");
```

Comment:
This assumption is incorrect. Base vector in a ConstantVector can have any size.